### ISSUE TITLE
Non-Cygwin sessions on Windows

### ISSUE THREAD

**Issue by nigels-com** [sedwards2009/extraterm#19]
My default multi-tab terminal app on Windows is ConEmu, which I find quite solid, although somewhat quirky.  One feature I do make use of is the various _tasks_ that can be configured - such as `cmd.exe`, git bash, MSYS or MSYS2, or a direct remote ssh session.  I don't mind omitting the native windows shells, personally.  But I do have some preference for git bash or MSYS2, rather than Cygwin.  Looking at main.ts this looks like a matter of adding some alternative enumerants, along with the logic to either auto-detect the various options, or require pull paths as a last resort.

But I do wonder if it's possible to spawn these other shells without inheriting Cygwin environment via extraterm?

![conemu](https://cloud.githubusercontent.com/assets/545677/14226005/c09f1e46-f919-11e5-9a39-43098e11adb1.png)

**Comment by sedwards2009** [comment]
The main problem is that I don't know how to run those commands correctly such that they run in proper interactive mode. The cygwin support uses a python process as a proxy to get a foot into the cygwin world where we can use the PTY support inside cygwin's Python to start the shell process.

I imagine something similar is needed here too.

**Comment by ppklewczuk** [comment]
I didn't create a separate issue, so I dig old issue - has something has changed over that time? Have you thought about supporting WSL?

**Comment by sedwards2009** [comment]
Nothing has changed, yet. In the last month I've been thinking about it and it has become very clear that Extraterm needs better support for different session types, and also for  the different types which you have on Windows like cmd, powershell, cygwin and also WSL. Especially WSL. For a lot of people it is the an obvious replacement for cygwin which also works much better.

I've also been looking into PowerShell and reading up on it. It is now much more interesting because it has become open source and cross platform. It's approach to shuffling objects around is a great match to the kinds of features I want in Extraterm.

I want to work on this before starting the big job of replacing how the text rendering is done. So hopefully it won't be long.

**Comment by sedwards2009** [comment]
This issue is resolved in the 0.35 release.
